### PR TITLE
Add "undefined" to highlight words in log parser

### LINKED ISSUE #2783: 'undefined' should be highlighted in the the gubernator display
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/45903/pull-kubernetes-unit/30908/build-log.txt

gives an example of a hard to analyze output, the key text is:

```
W0516 12:44:55.020] # k8s.io/kubernetes/pkg/api/testing
W0516 12:44:55.021] pkg/api/testing/fuzzer.go:429: undefined: AzureDataDiskKind
W0516 12:44:55.021] pkg/api/testing/fuzzer.go:430: undefined: AzureSharedBlobDisk
```

But it wasn't highlighted.

### PR DESCRIPTION
**Pull request by cblecker**
Fixes #2783.

Also fixes spelling of "highlight".

### CODE REVIEW

File: gubernator/log_parser.py [hunk 1]
```diff
@@ -107,17 +107,18 @@ def digest(data, objref_dict=None, filters=None, error_re=regex.error_re,
     if filters is None:
         filters = {'Namespace': '', 'UID': '', 'pod': '', 'ContainerID':''}
 
-    hilight_words = ["error", "fatal", "fail", "failed", "build timed out"]
+    highlight_words = ["error", "fatal", "fail", "failed", "build timed out",
```
Comment (fejta) [inline gubernator/log_parser.py]: Please put a line break after `[`

File: mungegithub/bulk-lgtm/www/angular-material.js [hunk 2]
```diff
@@ -20723,7 +20723,7 @@ MdChipsCtrl.prototype.hasFocus = function () {
    *
    *   <ul>Validation
    *     <li>allow a validation callback</li>
-   *     <li>hilighting style for invalid chips</li>
```
Comment (rmmh) [inline mungegithub/bulk-lgtm/www/angular-material.js]: probably don't need to modify the vendored code here, but whatever

File: gubernator/log_parser.py [hunk 3]
```diff
@@ -107,17 +107,18 @@ def digest(data, objref_dict=None, filters=None, error_re=regex.error_re,
...
+                       "undefined"]
```
Comment (fejta) [inline gubernator/log_parser.py]: This only captures undefined golang build errors. I think the re I mentioned in the issue will be more helpful

File: mungegithub/bulk-lgtm/www/angular-material.js (hunk 2 above)
Comment (cblecker) [inline mungegithub/bulk-lgtm/www/angular-material.js]: Ah! I didn't realize this was vendored pieces. Yeah, I'll remove these.

File: gubernator/log_parser.py (hunk 1 above)
Comment (cblecker) [inline gubernator/log_parser.py]: done

**Comment by cblecker** [comment]
/cc @cblecker
testing

**Comment by cblecker** [comment]
I added "errors" to the highlight list as well, as that word shouldn't come up unless there's an actual problem either.

**Comment by cblecker** [comment]
@fejta @rmmh: does this look good to you guys? I don't think we're quite at the point that an error regex will work with the variance in formatting between test scripts that run in go/python/bash.

Also, is there anything else I need to do as far as a version bump or anything for the gubernator? This is my first PR against it specifically.

**Comment by rmmh** [comment]
Please don't highlight "errors", it's almost entirely false positives. Undefined has mostly good highlights.

[undefined hits](http://storage.googleapis.com/k8s-gubernator/test/undefined_hits.txt) vs [errors hits](http://storage.googleapis.com/k8s-gubernator/test/errors_hits.txt).

**Comment by cblecker** [comment]
@rmmh Changed

**Comment by rmmh** [comment]
thanks!

**Pull request merged**